### PR TITLE
introduction of quarkus-deployment-bom

### LINKED ISSUE #3190: introduce quarkus-deployment-bom
Currently we have `quarkus-bom` which includes all the Quarkus extension runtime dependency versions. Which is then imported by `quarkus-build-parent` and user applications.

We are not offering an equivalent BOM for externally developed extension projects. A workaround would be to import `quarkus-build-parent` as a BOM. That's not its primary purpose though.

I suggest we introduce `quarkus-deployment-bom` (following the naming convention we established for extensions, i.e. extension-name -> extension-name-deployment) that would include all the Quarkus (Core) extension deployment dependency versions, which would also import `quarkus-bom` and then be imported by `quarkus-build-parent`.

### PR DESCRIPTION
**Pull request by aloubyansky**
Fixes #3190 

This change applies the extension artifact naming convention to the BOM(s) by moving `quarkus-bom` from `bom` dir to `bom/runtime` and introducing `quarkus-deployment-bom` under `bom/deployment`.

### CODE REVIEW

**Review by rsvoboda — COMMENTED** [review]
Should we have a doc related to BOMs in Quarkus?
Having multiple BOMs can make some people confused

**Comment by aloubyansky** [comment]
Definitely, I was going to update the docs in a separate PR. This change doesn't "break" the existing docs.

File: build-parent/pom.xml [hunk 1]
```diff
@@ -22,19 +22,10 @@
...
         <!-- These properties are needed in order for them to be resolvable by the documentation -->
         <!-- The Graal version we suggest using in documentation - as that's
            what we work with by self downloading it: -->
         <graal-sdk.version-for-documentation>19.0.2</graal-sdk.version-for-documentation>
-        <rest-assured.version>3.3.0</rest-assured.version>
-        <axle-client.version>0.0.7</axle-client.version>
-        <vertx.version>3.7.1</vertx.version>
-
-        <!-- Enable APT by default for Eclipse -->
-        <m2e.apt.activation>jdt_apt</m2e.apt.activation>
```
Comment (gsmet) [inline build-parent/pom.xml]: Mmmmh, where did you move these properties? I don't see them anymore (and they are needed)?

**Review by gsmet — CHANGES_REQUESTED** [review]
I added a question. I might have missed something but it seems wrong.

File: build-parent/pom.xml (hunk 1 above)
Comment (aloubyansky) [reply build-parent/pom.xml]: Ugh, my bad. I had another pom which I decided not to include here. Thanks for catching it @gsmet !

File: bom/runtime/pom.xml [hunk 2]
```diff
@@ -588,11 +606,13 @@
                 <groupId>io.quarkus</groupId>
                 <artifactId>quarkus-test-h2</artifactId>
                 <version>${project.version}</version>
+                <scope>test</scope>
             </dependency>
             <dependency>
                 <groupId>io.quarkus</groupId>
                 <artifactId>quarkus-test-kubernetes-client</artifactId>
                 <version>${project.version}</version>
+                <scope>test</scope>
```
Comment (gsmet) [inline bom/runtime/pom.xml]: We decided against adding the `scope` in the BOM and we explicitly removed them. It's considered bad practice.

Comment (aloubyansky) [reply bom/runtime/pom.xml]: Ok, removed scope test.

**Review by gsmet — APPROVED** [review]
Looks good to me now.

**Pull request merged**